code/projects/steg/src/lib/src/vc.rs

```diff
@@ -0,0 +1,202 @@
+use crate::image_data::{create_image_data, ImageDataExt, RGBA_CHANNELS};
+use tinyrand::{Rand, Seeded, StdRand};
+use wasm_bindgen::prelude::*;
+use web_sys::ImageData;
+
+const BLACK: u8 = u8::MIN;
+const WHITE: u8 = u8::MAX;
+
+#[wasm_bindgen]
+#[derive(Clone, Copy, Debug)]
+pub enum VcGrayscale {
+    Average = "average",
+    Luminosity = "luminosity",
+}
+
+#[wasm_bindgen]
+#[derive(Clone, Copy, Debug)]
+pub enum VcDithering {
+    Threshold = "threshold",
+    Random = "random",
+    FloydSteinberg = "floyd-steinberg",
+}
+
+fn closest_palette_color(v: u8) -> u8 {
+    if v < u8::MAX / 2 {
+        BLACK
+    } else {
+        WHITE
+    }
+}
```

Comment:
1. Thresholding could be done differently than just dividing the scale in half and quantizing to `MAX` or `MIN`.
2. nit: *Do we really need to make constants for `BLACK` and `WHITE`? Isn't it kinda obvious?*
If we need those two constants, why don't we create  `const HALFPOINT_THRESHOLD = u8::MAX / 2`? Maybe somebody does not understand what this means and needs a name for it.